### ISSUE TITLE
Fprime-gds GUI not handling NaNs

### ISSUE THREAD

**Issue by budhe888** [nasa/fprime-gds#123]
| | |
|:---|:---|
|**_F´ Version_**|  3.1.4 |
|**_Affected Component_**| fprime-gds loader.js  |
---
## Problem Description

When a telemetry channel sent to fprime-gds contains a value of `NaN`, it appears that the GUI does not know how to handle it, and so the telemetry stops updating in the Channels window. When looking at developer tools in chrome, I get the following error message:

```sh
VM251:1 Uncaught SyntaxError: Unexpected token 'N', ..."2, "val": NaN, "disp"... is not valid JSON
    at JSON.parse (<anonymous>)
    at xhttp.onreadystatechange (loader.js:198:40)
xhttp.onreadystatechange @ loader.js:198
XMLHttpRequest.send (async)
(anonymous) @ loader.js:216
load @ loader.js:191
poller @ loader.js:248
```

Inspecting the JSON that it complains about:
```json
{
  "time": {
    "base": 2,
    "context": 0,
    "seconds": 1711929600,
    "microseconds": 197687
  },
  "id": 8972,
  "val": NaN,
  "display_text": "nanC"
},
{
  "time": {
    "base": 2,
    "context": 0,
    "seconds": 1711929600,
    "microseconds": 197694
  },
  "id": 8973,
  "val": NaN,
  "display_text": "nanC"
},
```

I verified that fprime-gds actually receives the value just fine, as you can see below if I use the CLI to view the data.

```sh
$ fprime-cli channels -d ~/dev-working-tlm-power/CadreFlightDeploymentTopologyAppDictionary.xml -p 50058 -s nan
2024-04-01 00:08:11: power.PWR_BattExtTemp1 (8972) (2(0)-1711930091:106129) nanC
2024-04-01 00:08:11: power.PWR_BattExtTemp2 (8973) (2(0)-1711930091:106134) nanC
2024-04-01 00:08:12: power.PWR_BattExtTemp1 (8972) (2(0)-1711930092:156472) nanC
2024-04-01 00:08:12: power.PWR_BattExtTemp2 (8973) (2(0)-1711930092:156491) nanC
2024-04-01 00:08:13: power.PWR_BattExtTemp1 (8972) (2(0)-1711930093:199988) nanC
2024-04-01 00:08:13: power.PWR_BattExtTemp2 (8973) (2(0)-1711930093:199993) nanC
2024-04-01 00:08:14: power.PWR_BattExtTemp1 (8972) (2(0)-1711930094:253030) nanC
2024-04-01 00:08:14: power.PWR_BattExtTemp2 (8973) (2(0)-1711930094:253048) nanC
```

## How to Reproduce

1. Create an fprime component that sends telemetry channel with a value of `NaN`
2. Run fprime-gds with the GUI enabled
3. Open the web browser fprime-gds GUI and go to the channels tab
4. Run fprime deployment with above component
5. Observe that the GUI channels tab stops updating. 

## Expected Behavior
It should gracefully handle a NaN and probably just display `Nan` in the browser.

**Comment by LeStarch** [comment]
The issue comes with the JSON serialization in Python.  We should update that serialization to mark items as NaN in text and re-translate in the webui.

**Comment by LeStarch** [comment]
This was fixed with PR #134
